### PR TITLE
ORC-728: Support head command in Java Tools

### PR DESCRIPTION
**Pull request by yuokada**
### What changes were proposed in this pull request?

- Improve orc-tools command to support head command in Java Tools

### How was this patch tested?

- Confirmed as follows:

```
~/w/I/o/java ❯❯❯ java -jar tools/target/orc-tools-1.7.0-SNAPSHOT-uber.jar data --help t/
usage: java -jar orc-tools-*.jar data <orc file>*
 -h,--help            Provide help
 -n,--lines <LINES>

~/w/I/o/java ❯❯❯ java -jar tools/target/orc-tools-1.7.0-SNAPSHOT-uber.jar data --lines 3 t/output.orc
log4j:WARN No appenders could be found for logger (org.apache.hadoop.util.Shell).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.hadoop.security.authentication.util.KerberosUtil (file:/Users/yuokada/works/IdeaProjects/orc/java/tools/target/orc-tools-1.7.0-SNAPSHOT-uber.jar) to method sun.security.krb5.Config.getInstance()
WARNING: Please consider reporting this to the maintainers of org.apache.hadoop.security.authentication.util.KerberosUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Processing data file t/output.orc [length: 257]
{"time":"2019-01-01 00:00:00.0"}
{"time":"2015-01-01 00:00:00.0001"}
{"time":"2015-01-01 00:00:00.0"}
________________________________________________________________________________________________________________________

```

### CODE REVIEW

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you for making a PR, @yuokada .
However, could you add this new option `-n` to the existing `PrintData.java` instead of making a new `PrintHeadData.java`?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @yuokada .
I verified manually. Merged to master.

File: java/tools/src/java/org/apache/orc/tools/PrintData.java [hunk 1]
```diff
@@ -195,42 +200,82 @@ static void printRow(JSONWriter writer,
...
     RecordReader rows = reader.rows();
     try {
       TypeDescription schema = reader.getSchema();
       VectorizedRowBatch batch = schema.createRowBatch();
+      Integer counter = 0;
       while (rows.nextBatch(batch)) {
-        for(int r=0; r < batch.size; ++r) {
+        for (int r=0; r < batch.size; ++r) {
           JSONWriter writer = new JSONWriter(out);
           printRow(writer, batch, schema, r);
```
Comment (dongjoon-hyun) [inline java/tools/src/java/org/apache/orc/tools/PrintData.java:213]: Hi, @yuokada . Sorry, but I missed that this doesn't work correct in some files. Could you check this once more?
```
$ java -jar tools/target/orc-tools-1.7.0-SNAPSHOT-uber.jar data -n 1 ../examples/orc_no_format.orc 
log4j:WARN No appenders could be found for logger (org.apache.hadoop.util.Shell).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Processing data file ../examples/orc_no_format.orc [length: 745]
{"_col0":["Java\/C++\/Linux,4306:Bob\/4307:Jack,Bob\/90"],"_col1":null,"_col2":null}
________________________________________________________________________________________________________________________

java:master $ java -jar tools/target/orc-tools-1.7.0-SNAPSHOT-uber.jar data -n 1 ../examples/orc-file-11-format.orc 
log4j:WARN No appenders could be found for logger (org.apache.hadoop.util.Shell).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Processing data file ../examples/orc-file-11-format.orc [length: 373336]
{"boolean1":false,"byte1":1,"short1":1024,"int1":65536,"long1":9223372036854775807,"float1":1,"double1":-15,"bytes1":[0,1,2,3,4],"string1":"hi","middle":{"list":[{"int1":1,"string1":"bye"},{"int1":2,"string1":"sigh"}]},"list":[{"int1":3,"string1":"good"},{"int1":4,"string1":"bad"}],"map":[],"ts":"2000-03-12 15:00:00.0","decimal1":"12345678.6547456"}
{"boolean1":false,"byte1":1,"short1":1024,"int1":65536,"long1":9223372036854775807,"float1":1,"double1":-15,"bytes1":[0,1,2,3,4],"string1":"hi","middle":{"list":[{"int1":1,"string1":"bye"},{"int1":2,"string1":"sigh"}]},"list":[{"int1":3,"string1":"good"},{"int1":4,"string1":"bad"}],"map":[],"ts":"2000-03-12 15:00:00.0","decimal1":"12345678.6547456"}
{"boolean1":false,"byte1":1,"short1":1024,"int1":65536,"long1":9223372036854775807,"float1":1,"double1":-15,"bytes1":[0,1,2,3,4],"string1":"hi","middle":{"list":[{"int1":1,"string1":"bye"},{"int1":2,"string1":"sigh"}]},"list":[{"int1":3,"string1":"good"},{"int1":4,"string1":"bad"}],"map":[],"ts":"2000-03-12 15:00:00.0","decimal1":"12345678.6547456"}
{"boolean1":false,"byte1":1,"short1":1024,"int1":65536,"long1":9223372036854775807,"float1":1,"double1":-15,"bytes1":[0,1,2,3,4],"string1":"hi","middle":{"list":[{"int1":1,"string1":"bye"},{"int1":2,"string1":"sigh"}]},"list":[{"int1":3,"string1":"good"},{"int1":4,"string1":"bad"}],"map":[],"ts":"2000-03-12 15:00:00.0","decimal1":"12345678.6547456"}
{"boolean1":false,"byte1":1,"short1":1024,"int1":65536,"long1":9223372036854775807,"float1":1,"double1":-15,"bytes1":[0,1,2,3,4],"string1":"hi","middle":{"list":[{"int1":1,"string1":"bye"},{"int1":2,"string1":"sigh"}]},"list":[{"int1":3,"string1":"good"},{"int1":4,"string1":"bad"}],"map":[],"ts":"2000-03-12 15:00:00.0","decimal1":"12345678.6547456"}
{"boolean1":false,"byte1":1,"short1":1024,"int1":65536,"long1":9223372036854775807,"float1":1,"double1":-15,"bytes1":[0,1,2,3,4],"string1":"hi","middle":{"list":[{"int1":1,"string1":"bye"},{"int1":2,"string1":"sigh"}]},"list":[{"int1":3,"string1":"good"},{"int1":4,"string1":"bad"}],"map":[],"ts":"2000-03-12 15:00:00.0","decimal1":"12345678.6547456"}
{"boolean1":false,"byte1":1,"short1":1024,"int1":65536,"long1":9223372036854775807,"float1":1,"double1":-15,"bytes1":[0,1,2,3,4],"string1":"hi","middle":{"list":[{"int1":1,"string1":"bye"},{"int1":2,"string1":"sigh"}]},"list":[{"int1":3,"string1":"good"},{"int1":4,"string1":"bad"}],"map":[],"ts":"2000-03-12 15:00:00.0","decimal1":"12345678.6547456"}
{"boolean1":false,"byte1":1,"short1":1024,"int1":65536,"long1":9223372036854775807,"float1":1,"double1":-15,"bytes1":[0,1,2,3,4],"string1":"hi","middle":{"list":[{"int1":1,"string1":"bye"},{"int1":2,"string1":"sigh"}]},"list":[{"int1":3,"string1":"good"},{"int1":4,"string1":"bad"}],"map":[],"ts":"2000-03-12 15:00:00.0","decimal1":"12345678.6547456"}
________________________________________________________________________________________________________________________
```

Comment (yuokada) [reply java/tools/src/java/org/apache/orc/tools/PrintData.java:213]: @dongjoon-hyun Sorry for late response. I sent a new pull request for this issue.
https://github.com/apache/orc/pull/737

**Pull request merged**